### PR TITLE
updates node status

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1918.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -26,10 +26,29 @@
...
+  branch = "master"
+  name = "github.com/docker/distribution"
+
 [[constraint]]
   branch = "master"
   name = "github.com/giantswarm/apiextensions"
 
+[[override]]
+  branch = "waiting-for-connection"
+  name = "github.com/giantswarm/errors"
```
Comment (xh3b4sd) [inline Gopkg.toml]: See https://github.com/giantswarm/errors/pull/11.

File: create.go [hunk 2]
```diff
@@ -3,34 +3,116 @@ package statusresource
...
-	accessor, err := meta.Accessor(obj)
+	// TODO
+	//
+	// fetch new object
+	//
+	// compute patches
+	//
+	// Apply the computed list of patches to make the status update take effect.
+	// In case there are no patches we do not need to do anything here. So we
+	// prevent unnecessary API calls.
```
Comment (xh3b4sd) [inline create.go]: I still have to sort the godoc here.

File: resource.go [hunk 3]
```diff
@@ -12,9 +13,16 @@ const (
...
-	Logger            micrologger.Logger
-	NodeCountFunc     func(v interface{}) (int, error)
+	ClusterEndpointFunc func(v interface{}) (string, error)
+	ClusterIDFunc       func(v interface{}) (string, error)
+	ClusterStatusFunc   func(v interface{}) (providerv1alpha1.StatusCluster, error)
+	// TODO replace this with a G8sClient to fetch the node versions from the
+	// NodeConfig status once we can use the NodeConfig for general node
+	// management. As of now NodeConfig CRs are still used for draining in older
+	// guest clusters.
+	GuestCluster  guestcluster.Interface
```
Comment (xh3b4sd) [inline resource.go]: The todo maybe should be more explicit depending on what we want to have it look like now. The decision process might be still ongoing.

**Review by kopiczko — APPROVED** [review]

File: create.go [hunk 4]
```diff
@@ -95,40 +178,75 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+					return nil, microerror.Maskf(missingLabelError, labelVersion)
+				}
 
-	err = r.restClient.Patch(types.JSONPatchType).AbsPath(accessor.GetSelfLink()).Body(b).Do().Error()
-	if err != nil {
-		return microerror.Mask(err)
+				nodes = append(nodes, providerv1alpha1.StatusClusterNode{
+					Name:    n,
+					Version: v,
+				})
```
[the remark this answers is not in the document]
Comment (xh3b4sd) [reply create.go]: It is like you suggested there. Note that this here is also about the provider types and not the core types. Below we patch `/status/cluster/nodes`. So this is what you asked for in your comment over there.

[the remark this answers is not in the document]
Comment (kopiczko) [reply create.go]: Oh yeah. Good morning :)

[the remark this answers is not in the document]
Comment (xh3b4sd) [reply create.go]: 👋

[the remark this answers is not in the document]
Comment (kopiczko) [reply create.go]: I was confused. I get it now this is provider CR, not NodeConfig CR.

[the remark this answers is not in the document]
Comment (xh3b4sd) [reply create.go]: Yes. I think we should later just mirror the same thing in the provider CR when `node-operator` is providing the information.

**Comment by xh3b4sd** [comment]
Verified via Azure and AWS e2e tests.

**Pull request merged**